### PR TITLE
Lc39 refactor

### PR DESCRIPTION
**Pull request by ilteoood**
#### Checklist

- [X] your branch will not cause merge conflict with `master`
- [X] run `npm test`
- [X] tests are included
- [X] the documentation is updated or intregrated accordingly with your changes
- [X] the code will follow the lint rules and style of the project
- [X] you are not committing extraneous files or sensible data

This PR doesn't include new functionalities, but just little refactors.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -27,13 +27,8 @@ function parsePort(port) {
...
 
 function parseBoolean(bool) {
-  if (!bool) {
-    return false
-  }
-  if (bool === 'false' || bool === '0') {
-    return false
-  }
-  return Boolean(bool)
+  const nonBoolValue = !bool || bool === 'false' || bool === '0'
```
Comment (davidebianchi) [inline bin/cli.js:30]: The variable name is not clear. It is a bool value. And the check does not check if it is a non bool value.

The function as written before seems to me to be clearer than the refactor.

File: lib/custom-logger.js [hunk 2]
```diff
@@ -41,11 +41,7 @@ function pinoOptions(moduleOptions, options) {
 }
 
 function removePort(host) {
-  if (!host) {
-    return undefined
-  }
-  const hostname = host.split(':')
-  return hostname[0]
+  return !host ? undefined : host.split(':')[0]
```
Comment (davidebianchi) [inline lib/custom-logger.js:44]: Why all in one line function? 
Maybe the hostname could be rewritten to be inline in return, but I would maintain the check separated. The code is quite more readable.

File: lib/launch-fastify.js [hunk 3]
```diff
@@ -35,46 +35,51 @@ const lc39Options = {
...
+
 function validateLogLevel(logLevel) {
-  if (!logLevel) {
-    return
-  }
-  // we will check if the value pass in logLevel is one of the values accepted by pino.js
-  const regex = new RegExp('^(error|warn|info|debug|trace|silent)$')
-  if (regex.test(logLevel)) {
-    return
+  const isLogValid = isRegexValidated(logLevel, LOG_REGEX)
```
Comment (davidebianchi) [inline lib/launch-fastify.js:79]: isLogLevelValid

File: lib/options-extractors.js [hunk 4]
```diff
@@ -43,13 +43,9 @@ function exportServiceOptions(options) {
...
   }
 
-  if (envVariables) {
-    for (const key in envVariables) {
-      if ({}.hasOwnProperty.call(envVariables, key)) {
-        fastifyPluginOptions[key] = envVariables[key]
-      }
-    }
-  }
+  Object
```
Comment (davidebianchi) [inline lib/options-extractors.js:46]: Why this change?

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: lib/options-extractors.js (hunk 4 above)
Comment (ilteoood) [reply lib/options-extractors.js:46]: I thought that iterate an array is faster than call hasOwnProperty for each entry

File: lib/launch-fastify.js [hunk 5]
```diff
@@ -35,46 +35,51 @@ const lc39Options = {
...
+  // disable the global require rule for importing the right module from the user given file
+  // eslint-disable-next-line global-require
+  const importedModule = require(absolutePath(filePath))
+  checkAsyncFunction(importedModule)
+  checkParamsLength(importedModule)
   return importedModule
 }
 
+function isRegexValidated(value, regex) {
+  return !value || regex.test(value)
```
Comment (fredmaggiowski) [inline lib/launch-fastify.js:63]: If I'm correct, this change introduces a breaking change, in fact prior this change the `validatePrefix` function provided with a falsy `prefix` would simply return undefined, now it will throw an error instead.

I think this is not emerged because there may be some missing tests, can you please verify this change with a new test?

Comment (ilteoood) [reply lib/launch-fastify.js:63]: Hi @fredmaggiowski,
I've improved the tests of launch-fastify with also good prefixes example, but I can't reach the use case that you're thinking about.

File: lib/options-extractors.js (hunk 4 above)
Comment (davidebianchi) [reply lib/options-extractors.js:46]: It is not so true. In this way, the object is also converted in array. And the check is only a startup, so the performance are not so important.

**Review by davidebianchi — APPROVED** [review]

**Comment by davidebianchi** [comment]
LGTM

**Comment by ilteoood** [comment]
ping

File: lib/launch-fastify.js (hunk 5 above)
Comment (fredmaggiowski) [reply lib/launch-fastify.js:63]: I've seen the test and it sould legit, perfect; thanks!
